nsv13/code/controllers/subsystem/starsystem.dm

```diff
@@ -0,0 +1,199 @@
+	for(var/obj/structure/overmap/OM in GLOB.overmap_objects)
+		if(OM.main_overmap)
+			return OM
+
+/datum/controller/subsystem/starsystem/proc/find_main_miner()
+	for(var/obj/structure/overmap/OM in GLOB.overmap_objects)
+		if(OM.main_miner)
+			return OM
+
+/datum/controller/subsystem/starsystem/proc/find_system(obj/structure/overmap/OM) //Used to determine what system a ship is currently in. Famously used to determine the starter system that you've put the ship in.
+	var/datum/starsystem/found
+	for(var/datum/starsystem/S in systems)
+		for(var/thez in SSmapping.levels_by_trait(S.level_trait))
+			if(thez == OM.z)
+				found = S
+				break
+	return found
+
+/datum/controller/subsystem/starsystem/proc/modular_spawn_enemies(obj/structure/overmap/OM, datum/starsystem/target_sys)//Select Ship to Spawn and Location via Z-Trait
+	var/list/levels = SSmapping.levels_by_trait(target_sys.level_trait)
+	if(levels?.len == 1)
+		var/turf/exit = get_turf(locate(round(world.maxx * 0.5, 1), round(world.maxy * 0.5, 1), levels[1])) //Plop them bang in the center of the system.
+		var/turf/destination = get_turf(pick(orange(20,exit)))
+		var/obj/structure/overmap/enemy = new OM(destination)
+		target_sys.add_enemy(enemy)
+	else if(levels?.len > 1)
+		message_admins("More than one level found for [target_sys]!")
+		return
+	else
+		message_admins("No levels forund for [target_sys]!")
+		return
+
+///////BOUNTIES//////
+
+/datum/controller/subsystem/starsystem/proc/bounty_payout()
+	cycle_bounty_timer()
+	if(bounty_pool == 0) //No need to spam when there is no cashola payout
+		return
+	minor_announce("Bounty Payment Processed", "Naval Command") //Temp - replace words and methods
+	var/split_bounty = bounty_pool / 2 //Split between our two accounts
+	var/datum/bank_account/D = SSeconomy.get_dep_account(ACCOUNT_CAR)
+	D.adjust_money(split_bounty)
+	var/datum/bank_account/DD = SSeconomy.get_dep_account(ACCOUNT_MUN)
+	DD.adjust_money(split_bounty)
+	bounty_pool = 0
+
+/datum/controller/subsystem/starsystem/proc/cycle_bounty_timer()
+	addtimer(CALLBACK(src, .proc/bounty_payout), 15 MINUTES) //Cycle bounty payments every 15 minutes
+
+//////NEW GAMEPLAY LOOP///////
+
+/datum/controller/subsystem/starsystem/proc/cycle_gameplay_loop()
+	addtimer(CALLBACK(src, .proc/gameplay_loop), rand(10 MINUTES, 15 MINUTES))
+
+/datum/controller/subsystem/starsystem/proc/gameplay_loop() //A very simple way of having a gameplay loop. Every couple of minutes, the Syndicate appear in a system, the ship has to destroy them.
+	var/datum/starsystem/current_system //Dont spawn enemies where theyre currently at
+	for(var/obj/structure/overmap/OM in GLOB.overmap_objects) //The ship doesnt start with a system assigned by default
+		if(!OM.main_overmap)
+			continue
+		OM.current_system = find_system(OM) //The ship doesnt start with a system assigned by default
+		current_system = OM?.current_system
+	for(var/datum/starsystem/starsys in systems)
+		if(starsys != current_system)

```

Comment:
```suggestion
		if(starsys == current_system)
				continue		

```
(Optional) We can remove all these indents here to make it look neater with an early return